akka-bench-jmh/src/main/scala/akka/remote/artery/BenchTestSource.scala

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.stream.Attributes
+import akka.stream.Outlet
+import akka.stream.SourceShape
+import akka.stream.stage.GraphStage
+import akka.stream.stage.GraphStageLogic
+import akka.stream.stage.OutHandler
+
+class BenchTestSource(elements: Int) extends GraphStage[SourceShape[Int]] {
+
+  val out: Outlet[Int] = Outlet("BenchTestSource")
+  override val shape: SourceShape[Int] = SourceShape(out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with OutHandler {
+
+      var n = 0
+
+      override def onPull(): Unit = {
+        n += 1
+        if (n > elements)
+          complete(out)
+        else
+          push(out, n)
```

Comment:
I never used this stage in the end, I used `BenchTestSourceSameElement` instead, but sometimes it might be nice with a counter. Good idea to pre-allocate them in the constructor of the stage. I'll change to that.